app/javascript/components/rooms/RoomsList.jsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { Row, Col, Container, Button, Spinner, Form, FormControl } from "react-bootstrap";
+import RoomCard from "./RoomCard";
+import useRooms from "../../hooks/queries/rooms/useRooms";
+import ButtonLink from "../stylings/buttons/ButtonLink";
+
+export default function RoomsList() {
+
+  const { isLoading, data: rooms } = useRooms();
+  if (isLoading) return <Spinner />
+
+  return (
+    <>
+      <Container>
+        <Row className='mt-4 mb-4'>
+          <Col>
+            {/* TODO: May need to change search form when implementing search functionality */}
+            <Form className='d-flex'>
+              <FormControl
+                type='search'
+                placeholder='Search'
+                className=''
+                style={{ width: '18rem' }}
+                aria-label='Search' />
+              <Button className='ms-2' variant="outline-secondary">Search</Button>
+            </Form>
```

Comment:
I was looking briefly on how the searching would be done with React and I saw people using forms and doing the whole searching functionality with a form. I added that comment on top because it really depends on what approach is taken at the time